cmd/dep/status.go

```diff
@@ -362,82 +363,121 @@ func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceMana
+					prm, _ := ptr.ToReachMap(true, false, false, nil)
+					bs.Children = prm.FlattenFn(paths.IsStandardImportPath)
+				}
+
+				// Split apart the version from the lock into its constituent parts
+				switch tv := proj.Version().(type) {
+				case gps.UnpairedVersion:
+					bs.Version = tv
+				case gps.Revision:
+					bs.Revision = tv
+				case gps.PairedVersion:
+					bs.Version = tv.Unpair()
+					bs.Revision = tv.Revision()
 				}
-			}
 
-			// Only if we have a non-rev and non-plain version do/can we display
-			// anything wrt the version's updateability.
-			if bs.Version != nil && bs.Version.Type() != gps.IsVersion {
-				c, has := p.Manifest.Constraints[proj.Ident().ProjectRoot]
-				if !has {
-					c.Constraint = gps.Any()
+				// Check if the manifest has an override for this project. If so,
+				// set that as the constraint.
+				if pp, has := p.Manifest.Ovr[proj.Ident().ProjectRoot]; has && pp.Constraint != nil {
+					bs.hasOverride = true
+					bs.Constraint = pp.Constraint
+				} else {
+					bs.Constraint = gps.Any()
+					for _, c := range cm[bs.ProjectRoot] {
+						bs.Constraint = c.Intersect(bs.Constraint)
+					}
 				}
-				// TODO: This constraint is only the constraint imposed by the
-				// current project, not by any transitive deps. As a result,
-				// transitive project deps will always show "any" here.
-				bs.Constraint = c.Constraint
-
-				vl, err := sm.ListVersions(proj.Ident())
-				if err == nil {
-					gps.SortPairedForUpgrade(vl)
-
-					for _, v := range vl {
-						// Because we've sorted the version list for
-						// upgrade, the first version we encounter that
-						// matches our constraint will be what we want.
-						if c.Constraint.Matches(v) {
-							bs.Latest = v.Revision()
-							break
+
+				// Only if we have a non-rev and non-plain version do/can we display
+				// anything wrt the version's updateability.
+				if bs.Version != nil && bs.Version.Type() != gps.IsVersion {
+					c, has := p.Manifest.Constraints[proj.Ident().ProjectRoot]
+					if !has {
+						c.Constraint = gps.Any()
+					}
+					// TODO: This constraint is only the constraint imposed by the
+					// current project, not by any transitive deps. As a result,
+					// transitive project deps will always show "any" here.
+					bs.Constraint = c.Constraint
+
+					vl, err := sm.ListVersions(proj.Ident())
+					if err == nil {
+						gps.SortPairedForUpgrade(vl)
+
+						for _, v := range vl {
+							// Because we've sorted the version list for
+							// upgrade, the first version we encounter that
+							// matches our constraint will be what we want.
+							if c.Constraint.Matches(v) {
+								bs.Latest = v.Revision()
+								break
+							}
 						}
 					}
 				}
-			}
 
-			out.BasicLine(&bs)
+				bsCh <- &bs
+
+			}(proj)
 		}
+
+		wg.Wait()
+		close(bsCh)
+		close(errorCh)
 		logger.Println()
+
+		if len(errorCh) > 0 {
+			for err := range errorCh {
+				ctx.Err.Println(err.Error())
+			}
+			ctx.Err.Println()
```

Comment:
I didn't really consider if it should still return an error. Could this look like a false positive if a user overlooks the logged errors? Should we return an error, and adjust the caller's error handling to print the buffer followed by a clear error message?

Reply:
Even that would work. Like conditionally print the out buffer (with incomplete info) depending on the error type. Although, I'm still not sure if that's the best behavior.

Reply:
Ah right, some errors are more severe than others.

What if errored items were still displayed inline, but indicated error somehow? (probably in addition to separate logs, since we don't want to cram messages into the table). Then there would be no mistaking the tail of the output for a false positive, with the errors interleaved. 

Or perhaps simplest is best, what if the error output just came last?